### ISSUE TITLE
Maintenance & Future of Seed

### ISSUE THREAD

**Issue by flosse** [David-OConnor/seed#672]
It was a pleasure for me to be working on the release of `v0.9.0` :smile:
And I would have great desire to develop Seed further, however, it is unfortunately as so often with open source projects, that there is no funding for it and therefore for me is not possible.
As far a I know the same is true for @David-OConnor, @MartinKavik, @TatriX, @arn-the-long-beard, @glennsl, @fosskers and all other contributors.

From there, I have an open question for the community: 
**How should we deal with the situation?**

A few off-the-cuff ideas:
- Include a prominent notice in the `README` and to the website
  - that this project is looking for a new maintainer
  - that this project is no longer maintained
  - that a further development needs $X thousand dollars
- Find companies that use Seed and are willing to donate a little bit

Or to look from an even higher bird's eye view:
Combine the efforts of all similar [open source frameworks](https://github.com/flosse/rust-web-framework-comparison#frontend-frameworks-wasm) like
- [Yew](https://yew.rs/)
- [sauron](https://github.com/ivanceras/sauron)
- etc.

One could e.g. develop shared crates like a `super-awesome-vdom` or so.

I would be very happy to hear about your thoughts and ideas!

**Comment by MartinKavik** [comment]
> One could e.g. develop shared crates like a super-awesome-vdom or so.

There is [gloo](https://github.com/rustwasm/gloo). Both `Seed` and `Yew` use it under the hood. But it's pretty difficult to have shared libraries because Rust FE frameworks have pretty different goals/apis - for example `gloo` isn't integrated in `MoonZoon` and only basic crates like `web-sys` and `js-sys` are shared.

> Find companies that use Seed and are willing to donate a little bit

I've written mails to relatively many persons/companies that starred the Seed repo some months/years ago. There were some freelancing opportunities but from startups / smaller companies that can't really support the community and development financially. I can imagine it could be a bit better now thanks to Rust adoption by larger companies and higher number of Rust devs and Rust popularity in the web3 world.

I've also sent mails to some companies with known Rust OS projects support. But non-Rust frontend wasn't as painful as other non-Rust parts of their stack / business so there wasn't a big motivation to invest money and time to change the status quo and fire/teach their React devs Rust. However, it looks like the number of companies wanting Rust-only stack is slowly growing. 

Also the Elm architecture and unimpressive speed in some cases and missing SSR make the selling Seed to JS teams even more difficult task. I was trying to resolve these problems but "accidentally" created MoonZoon instead.

> that this project is no longer maintained

It's not super-actively developed these days but all basic features should be already implemented and bugs fixed so there is no need to rush the development to make the framework usable. So I would describe it as something like "passively maintained".
There is a lot of things that could be improved but that's a different story, it's already production-ready imho.

> that this project is looking for a new maintainer

It could be useful as a part of a "campaign" to find new active maintainers on Reddit, GitHub, This Week in Rust, etc.

---

- There are some ideas and links related to OS funding experiments: https://github.com/OpenHope/OpenHope
- The similar funding problems appear also in the research - see for example https://nadia.xyz/science-funding (the author has written many articles about OS and Research). Another interesting OS project where _"research debt"_ is mentioned: https://github.com/magmide/magmide
- Feel free redirect or remove the sponsor button (otherwise I'll do it when I have some time in the future).
- There are some blockchain things mentioned in the article linked above. I co-founded a company and our first app (written in Rust of course) will be a crowdfunding app based on a blockchain and MoonZoon. I hope it'll help to support also many OS projects as a replacement for services like Patreon because they are pretty costly / difficult to manage for both maintainers and backers from my experience. The idea is I want to create an ecosystem that would be able to support OS developers and researchers so they can focus on their projects instead of marketing or begging for money and burn out during the process how  I burned out.

**Comment by David-OConnor** [comment]
For some context, I now do my web pages and apps in HTML, CSS, with targetted JS as required for interactions. Much faster page loading, UI interaction etc than any framework, and easier to manage without config, dependencies etc. This used to not be viable, but HTML, CSS, and JS have come a long way.

**Comment by fosskers** [comment]
My own personal website is also purely HTML/CSS for performance/minimalism reasons, but I recognize the value of TEA apps. I'm general in favour of more shared underpinnings (i.e. crates) or a "one true framework to rule them all", but of course that might fragment our efforts as joked about in that well-known XKCD comic.

In terms of my own time commitments to Seed, I can mostly "keep an eye" on it unless my clients are willing to fund it. I will consult with them on that.

**Comment by flosse** [comment]
My website is also pure HTML/CSS :stuck_out_tongue_closed_eyes: but a *classical* website has completely different requirements than a full features **application** that runs in the browser, so we should not compare apples with pears!

**Comment by fosskers** [comment]
Precisely. TEA and Seed have proven quite useful for Browser Extensions.

**Comment by Toasterson** [comment]
Being a Maintainer in OpenIndiana I can safely say, unless you invade a org through working there you wont get support there :) We tried for 10 years only thing that really worked is use it as your stack of choice and scratch your itch. We Maintainers of illumos and OpenIndiana made the stack our own and also maintain it in a Passive/Active fashion. It's fun. But not for the code more for the people you meet along the way. And as long as there are no barriers to entry like the closure thing just now. It will grow. In time. Most important thing is to enjoy the work you make with it and learn from what it is used for and why you want it. Maybe MoonZoon/Yew and Seed have different purposes. TEA is for me simplistic and not needing to learn another language like elm helps me write the apps I need to write.

Hope this gives some insights
-Till

**Comment by ThoFrank** [comment]
Hi there,
Random dude here, who plays with Seed in his free time and just stumbled across this issue.
I watched this talk a while ago and it might also be interesting to you:
https://youtu.be/M2slgOo2jQ4

If I recall correctly, the EU basically has (or at least had) a budget that was used to pay open source developers. You would have to write a proposal about a specific project and be an EU citizen though. 

Thanks for your great work!

**Comment by stappersg** [comment]
MartinKarvik commented on Apr 8, 2022

> I was trying to resolve these problems but "accidentally" created MoonZoon instead.

Link to [MoonZoon](https://github.com/MoonZoon/MoonZoon).
